python/spec/fixtures/pipfile_files/unsupported_dep

```diff
@@ -1,8 +0,0 @@
-[[source]]
-name = "pypi"
-url = "https://pypi.org/simple"
-verify_ssl = true
-
-[packages]
-requests = "==2.18.0"
-pyobjc = "*"
```

Comment:
How does new pipenv handle this file (and lockfile) now?

Reply:
It fails with a generic `Dependabot::DependencyFileNotResolvable` but without a message explaining the dep can't be built on linux. There is not really any information to let us infer this is the case anymore, and the exception was specific to a single dependency

Reply:
`Dependabot::DependencyFileNotResolvable` is a Dependabot error, not specific to pipenv, right? My question was more, what message does new pipenv give now?

The previous message was not really related to platforms either, I guess it was impossible to resolve that particular dependency on Linux and we simply rescued the generic error and explained that.

Anyways, I was just curious. As you point out, it's a single dependency and worst case scenario would be that Dependabot gets enabled on a repo using it and the error message the user faces is worse now. Does not seem like a big deal!

Reply:
Ah, this is the error:


```
Creating a virtualenv for this project...
Pipfile: dependabot_tmp_dir/Pipfile
Using default python from /usr/local/.pyenv/versions/3.11.5/bin/python (3.11.5) to create virtualenv...
created virtual environment CPython3.11.5.final.0-64 in 432ms
  creator CPython3Posix(dest=/home/dependabot/.local/share/virtualenvs/dependabot_20230921-53449-6wdxjb-Z2f_DZo1, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/dependabot/.local/share/virtualenv)
    added seed packages: pip==23.2.1, setuptools==68.2.0, wheel==0.41.2
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment!
Virtualenv location: /home/dependabot/.local/share/virtualenvs/dependabot_20230921-53449-6wdxjb-Z2f_DZo1
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed!

INFO:pipenv.patched.pip._internal.operations.prepare:Collecting requests<=2.18.4,>=2.18.0 (from -r /tmp/pipenv-n0h_9viw-requirements/pipenv-gpwo7a4a-constraints.txt (line 3))
INFO:pipenv.patched.pip._internal.network.download:Using cached requests-2.18.4-py2.py3-none-any.whl (88 kB)
INFO:pipenv.patched.pip._internal.operations.prepare:Collecting pyobjc==5.1.1 (from -r /tmp/pipenv-n0h_9viw-requirements/pipenv-gpwo7a4a-constraints.txt (line 2))
INFO:pipenv.patched.pip._internal.network.download:Using cached pyobjc-5.1.1-py3-none-any.whl (2.8 kB)
INFO:pipenv.patched.pip._internal.operations.prepare:Collecting pyobjc-core==5.1.1 (from pyobjc==5.1.1->-r /tmp/pipenv-n0h_9viw-requirements/pipenv-gpwo7a4a-constraints.txt (line 2))
INFO:pipenv.patched.pip._internal.network.download:Using cached pyobjc-core-5.1.1.tar.gz (793 kB)
INFO:pipenv.patched.pip._internal.cli.spinners:Preparing metadata (setup.py): started
INFO:pipenv.patched.pip._internal.cli.spinners:Preparing metadata (setup.py): finished with status 'error'
ERROR:pip.subprocessor:[present-rich] python setup.py egg_info exited with 1
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/resolver.py", line 646, in _main
[ResolutionFailure]:       resolve_packages(
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/resolver.py", line 613, in resolve_packages
[ResolutionFailure]:       results, resolver = resolve(
[ResolutionFailure]:       ^^^^^^^^
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/resolver.py", line 593, in resolve
[ResolutionFailure]:       return resolve_deps(
[ResolutionFailure]:       ^^^^^^^^^^^^^
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 845, in resolve_deps
[ResolutionFailure]:       results, hashes, internal_resolver = actually_resolve_deps(
[ResolutionFailure]:       ^^^^^^^^^^^^^^^^^^^^^^
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 618, in actually_resolve_deps
[ResolutionFailure]:       resolver.resolve()
[ResolutionFailure]:   File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 444, in resolve
[ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  You can use $ pipenv run pip install <requirement_name> to bypass this mechanism, then run $ pipenv graph to inspect the versions actually installed in the virtualenv.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: metadata generation failed

Traceback (most recent call last):
  File "/usr/local/.pyenv/versions/3.11.5/bin/pipenv", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/cli/options.py", line 58, in main
    return super().main(*args, **kwargs, windows_expand_args=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/vendor/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/cli/command.py", line 340, in lock
    do_lock(
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/routines/lock.py", line 65, in do_lock
    venv_resolve_deps(
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 786, in venv_resolve_deps
    c = resolve(cmd, st, project=project)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/.pyenv/versions/3.11.5/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 655, in resolve
    raise RuntimeError("Failed to lock Pipfile.lock!")
RuntimeError: Failed to lock Pipfile.lock!
```

Reply:
We could actually maybe look for `Collecting pyobjc` I suppose

Reply:
I suppose so, yes. I think the aggressive removal is fine too though, there's probably bigger fish to fry.